doc/api/module.md

```diff
@@ -80,6 +80,96 @@ isBuiltin('fs'); // true
 isBuiltin('wss'); // false
 ```
 
+### `module.registerLoader()`
+
+<!-- YAML
+added: v20.x.x
+-->
+
+In addition to using the `--experimental-loader` option in the CLI,
+loaders can be registered programmatically using the `module.registerLoader()`
+method.
+
+```js
+import { registerLoader } from 'node:module';
+
+registerLoader('http-to-https');
+
+// Loader hooks haven't been executed yet.
+
+await import('./my-app.mjs'); // Same behavior of CLI: hooks will be executed before importing the module.
+```
+
+In the example above, we are registering the `http-to-https` loader,
+but it will only be available in the next imported module—in this
+case, in `my-app.mjs`.
+
+The `--experimental-loader` flag of the CLI could be used together
+the `registerLoader` function, the loaders registered with the
+function will follow the same execution chain of loaders registered
+within the CLI:
+
+```console
+node \
+  --experimental-loader unpkg \
+  --experimental-loader http-to-https \
+  --experimental-loader cache-buster \
+  entrypoint.mjs
+```
+
+```js
+// entrypoint.mjs
+import { URL } from 'node:url'
+import { registerLoader } from 'node:module'
+
+const loaderPath = new URL('./my-programmatically-loader.mjs', import.meta.url).href
+
+registerLoader(loaderPath)
+await import('./my-app.mjs')
+```
+
+The `my-programmatically-loader.mjs` could use all the resources
+provided by `unpkg`, `http-to-https` and `cache-buster` loaders.
+
+It's also possible to use `registerLoader` more than once:
+
+```js
+// entrypoint.mjs
+import { URL } from 'node:url'
+import { registerLoader } from 'node:module'
+
+const firstLoader = new URL('./first-loader.mjs', import.meta.url).href
+const secondLoader = new URL('./second-loader.mjs', import.meta.url).href
+
+registerLoader(firstLoader)
+registerLoader(secondLoader)
+await import('./my-app.mjs')
+```
+
+Both loaders (`first-loader.mjs` and `second-loader.mjs`) can use all
+the resources provided by the loaders registered in the CLI. But remember
+that they will only be available in the next imported module (`my-app.mjs`).
+The execution order of the hooks when importing `my-app.mjs` and consecutive
+modules in the example above will be:
+
+```console
+resolve: second-loader.mjs
+resolve: first-loader.mjs
+resolve: cache-buster
+resolve: http-to-https
+resolve: unpkg
+load: second-loader.mjs
+load: first-loader.mjs
+load: cache-buster
+load: http-to-https
+load: unpkg
+globalPreload: second-loader.mjs
+globalPreload: first-loader.mjs
+globalPreload: cache-buster
+globalPreload: http-to-https
+globalPreload: unpkg
+```
```

Comment:
I would combine this example with the previous

Reply:
Also, this is a wee bit redundant—maybe point to the ESM doc for an explanation on sequence?

Reply:
> I would combine this example with the previous

Do you mean join them in the same code block? Like this?:

```js
// entrypoint.mjs
import { URL } from 'node:url';
import { registerLoader } from 'node:module';

const firstLoader = new URL('./first-loader.mjs', import.meta.url).href;
const secondLoader = new URL('./second-loader.mjs', import.meta.url).href;

registerLoader(firstLoader);
registerLoader(secondLoader);
await import('./my-app.mjs');

// resolve: second-loader.mjs
// resolve: first-loader.mjs
// resolve: cache-buster
// resolve: http-to-https
// resolve: unpkg
// load: second-loader.mjs
// load: first-loader.mjs
// load: cache-buster
// load: http-to-https
// load: unpkg
// globalPreload: second-loader.mjs
// globalPreload: first-loader.mjs
// globalPreload: cache-buster
// globalPreload: http-to-https
// globalPreload: unpkg
```

We could point it to ESM doc but at least for me, It feels easier to understand the behavior of `registerLoader()` using the same modules (`second-loader.mjs`, `chache-buster`, etc) to create the examples. I think I could try to use only two or three loaders to get a smaller example, what do you think?